framework/Samplers/MCMC/MCMC.py

```diff
@@ -0,0 +1,293 @@
+                           'uniform': Distributions.Uniform(-1.0, 1.0)} # available proposal distributions
+    self._acceptDist = Distributions.Uniform(0.0, 1.0) # uniform distribution for accept/rejection purpose
+    # assembler objects
+    self.addAssemblerObject('proposal', '-n', True)
+
+  def localInputAndChecks(self, xmlNode, paramInput):
+    """
+      unfortunately-named method that serves as a pass-through for input reading.
+      comes from inheriting from Sampler and _readMoreXML chain.
+      @ In, xmlNode, xml.etree.ElementTree.Element, xml element node (don't use!)
+      @ In, paramInput, InputData.ParameterInput, parameter specs interpreted
+      @ Out, None
+    """
+    # this is just a passthrough until sampler gets reworked or renamed
+    self.handleInput(paramInput)
+
+  def handleInput(self, paramInput):
+    """
+      Read input specs
+      @ In, paramInput, InputData.ParameterInput, parameter specs interpreted
+      @ Out, None
+    """
+    likelihood = paramInput.findFirst('likelihood')
+    if likelihood is not None:
+      self._likelihood = likelihood.value
+    else:
+      self.raiseAnError(IOError, "likelihood is required, but not provided!")
+    init = paramInput.findFirst('samplerInit')
+    if init is not None:
+      # limit
+      limit = init.findFirst('limit')
+      if limit is not None:
+        self.limit = limit.value
+      else:
+        self.raiseAnError(IOError, 'MCMC', self.name, 'needs the limit block (number of samples) in the samplerInit block')
+      # initialSeed
+      seed = init.findFirst('initialSeed')
+      if seed is not None:
+        self.initSeed = seed.value
+      else:
+        self.initSeed = randomUtils.randomIntegers(0,2**31,self)
+      tune = init.findFirst('tune')
+      if tune is not None:
+        self._tune = tune.value
+    else:
+      self.raiseAnError(IOError, 'MCMC', self.name, 'needs the samplerInit block')
+    # variables additional reading
+    for varNode in paramInput.findAll('variable'):
+      var = varNode.parameterValues['name']
+      initNode = varNode.findFirst('initial')
+      if initNode:
+        self._initialValues[var] = initNode.value
+      else:
+        self._initialValues[var] = None
+      proposal = varNode.findFirst('proposal')
+      if proposal:
+        self._proposal[var] = proposal.value
+      else:
+        self._proposal[var] = None
+    # TargetEvaluation Node (Required)
+    targetEval = paramInput.findFirst('TargetEvaluation')
+    self._targetEvaluation = targetEval.value
+    self._updateValues = copy.copy(self._initialValues)
+
+  def initialize(self, externalSeeding=None, solutionExport=None):
+    """
+      This function should be called every time a clean MCMC is needed. Called before takeAstep in <Step>
+      @ In, externalSeeding, int, optional, external seed
+      @ In, solutionExport, DataObject, optional, a PointSet to hold the solution
+      @ Out, None
+    """
```

Comment:
This is a generic question: does this MCMC method allow the user to use a chain generated in the past that needs to be augmented?

Reply:
I think the answer for this question is No for current implementation. This seems to me similar to Restart option.

Reply:
No restart?

Reply:
Can you try a test for it?